compliance/src/main/java/bio/terra/pearl/compliance/SyncVantaUsers.java

```diff
@@ -0,0 +1,296 @@
+                        try {
+                            Thread.sleep(Duration.ofSeconds(delayInSeconds));
+                        } catch (InterruptedException e) {
+                            log.error("Interrupted while sleeping in response to rate limit ");
+                        }
+                        return res.bodyToMono(String.class).map(msg -> new RateLimitException(msg, delayInSeconds));
+                    })
+                    .bodyToMono(type)
+                    .retryWhen(Retry.withThrowable(throwableFlux -> {
+                        return throwableFlux.filter(t -> t instanceof RateLimitException).map(t -> {
+                            RateLimitException re = (RateLimitException) t;
+                            return Retry.fixedDelay(5, re.getRetryAfterDelayDuration());
+                        });
+                    })).block();
+            cursorCount.incrementAndGet();
+            results.set(response.getResults());
+
+            if (response != null) {
+                if (response.getResults() != null) {
+                    elements.get().addAll(response.getResults().getData());
+                }
+            } else {
+                break;
+            }
+        }
+        return elements.get();
+    }
+
+    private WebClient getWebClientForIntegration(String accessToken, String integrationId, String resourceKind) {
+        WebClient.Builder webClientBuilder = WebClient.builder().defaultHeaders(h -> {
+            h.setBearerAuth(accessToken);
+            h.setContentType(MediaType.APPLICATION_JSON);
+        });
+
+        return webClientBuilder.baseUrl(vantaBasePath + "v1/integrations/" + integrationId + "/resource-kinds/" + resourceKind + "/resources").build();
+    }
+
+    public String syncVantaAccounts() {
+        StringBuilder summary = new StringBuilder();
+        WebClient wc = WebClient.builder().baseUrl(vantaBasePath + "oauth/token").build();
+        AccessToken vantaToken= wc.post().bodyValue(new VantaCredentials("client_credentials", "vanta-api.all:read vanta-api.all:write", vantaClientId, vantaSecret))
+                .header("Content-Type", "application/json").retrieve().bodyToMono(AccessToken.class).block();
+        WebClient.builder().defaultHeaders(h -> {
+            h.setBearerAuth(vantaToken.getAccess_token());
+            h.setContentType(MediaType.APPLICATION_JSON);
+        });
+
+        // go through each integration and find scope changes that need to be made.
+        int resourceBatchSize = 50;
+        Collection<VantaObject> vantaObjects = new HashSet<>();
+        List<VantaIntegration> integrationsToSync = new ArrayList<>();
+        integrationsToSync.add(new VantaIntegration("jira", "JiraAccount", new ParameterizedTypeReference<VantaResultsResponse<JiraAccount>>() {}, Set.of("669a8422865ac5731466a323", "66c629d278231843aec354ff", "669a8431865ac5731469ed44",
```

Comment:
the Sets in these lines could use to be pulled into named constants at the tope of the file, e.g. "JIRA_ACCOUNT_IDS" with a comment about how to find them.  I imagine these sets will be the part of this code most often needing to be updated

Reply:
I think I will move these values into the secret so that we can change them without rebuilding/redeploying.  Will also add a few comments for how to discover them in the future.